### PR TITLE
COMMON: Add a mode to the INI parser to ignore garbage in some files

### PR DESCRIPTION
**Pull request by nevernever69**
The pull request resolve a bug in French sheila(https://bugs.scummvm.org/ticket/13920)
in INI Parser

- Added a  flag  strictParser Do not allow garbage to be present in the file (default true)
- Added a condition to check for Non-Ascii

### CODE REVIEW

**Comment by lephilousophe** [comment]
As this is a new feature, it should be disabled by default and enabled only when needed.

**Comment by sluicebox** [comment]
There are two bugs in this ticket:

1. Script parsing called `error`
2. `error` segfaulted instead of displaying the debugger with the error message

I just fixed the second one: 02682424609b74ab286fd8ce4aac0bb81f512e4d

File: common/formats/ini-file.h [hunk 1]
```diff
@@ -105,10 +105,18 @@ class INIFile {
...
+	/**
+	 * Load configuration from a file.
+	 *
+	 * @param filename     Name of an INI file to parse
+	 * @param strictParser Do not allow garbage to be present in the file (default true)
+	 *
+	 * @return  True if file was parsed successfully
+	 */
+	bool	loadFromFile(const String &filename, bool strictParser = true); /*!< Load configuration from a file. */
+	bool	loadFromFileOrDataFork(const String &filename, bool strictParser = true); /*!< Load configuration from a file in MacBinary format. */
```
Comment (sev-) [inline common/formats/ini-file.h:117]: Please add the documentation  here too

File: common/formats/ini-file.cpp [hunk 2]
```diff
@@ -198,6 +203,9 @@ bool INIFile::loadFromStream(SeekableReadStream &stream) {
 			section.keys.push_back(kv);
 		}
 	}
+	if(nonAsciilineCount) {
```
Comment (sev-) [inline common/formats/ini-file.cpp:206]: It must follow our code formatting

File: common/formats/ini-file.cpp [hunk 3]
```diff
@@ -198,6 +203,9 @@ bool INIFile::loadFromStream(SeekableReadStream &stream) {
...
+		warning("Ignored Total %d non-ASCII Lines.", nonAsciilineCount);
```
Comment (sev-) [inline common/formats/ini-file.cpp:207]: "loadFromStream(): %d lines with non-ASCII garbage ignored"

File: common/formats/ini-file.cpp [hunk 4]
```diff
@@ -115,6 +116,10 @@ bool INIFile::loadFromStream(SeekableReadStream &stream) {
 
 		if (line.size() == 0) {
 			// Do nothing
+		} else if (strictParser && !Common::isPrint(line[0])) {
+			// Non-ASCII character at the beginning of the line, complain and count lines
+			nonAsciilineCount++;
+			warning("Ignored non-ASCII Line %d.", lineno);
```
Comment (sev-) [inline common/formats/ini-file.cpp:122]: Please, we discussed like 30 times. NO MULTIPLE WARNINGS. Remove this line

**Review by sev- — CHANGES_REQUESTED** [review]
Added my comments

**Comment by sluicebox** [comment]
I have proposed an alternative approach at: https://github.com/scummvm/scummvm/pull/5511

I don't think ASCII characters are a problem at all, that just happens to be a pattern that matches this specific instance. The broader problem is that the INI parser accepts lines without delimiters. That is also a problem that's simpler to address.

**Comment by elasota** [comment]
I think @sluicebox's approach is better in this case, we already have two functions in INIFile to configure it to accept unusual INI file behavior (`allowNonEnglishCharacters` and `suppressValuelessLineWarning`) so I think it would be better to stick to that pattern instead of adding a new parameter.

File: common/formats/ini-file.h (hunk 1 above)
Comment (nevernever69) [reply common/formats/ini-file.h:117]: After every loadFromXXX function should add the documentation

**Comment by sev-** [comment]
Thank you, merging

**Pull request merged**